internal/services/accrual.go

```diff
@@ -0,0 +1,212 @@
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"net/http"
+	"strconv"
+	"time"
+
+	"github.com/Renal37/go-musthave-diploma-tpl/internal/database"
+	"github.com/Renal37/go-musthave-diploma-tpl/internal/logger"
+	"github.com/Renal37/go-musthave-diploma-tpl/internal/models"
+	"go.uber.org/zap"
+)
+
+// AccrualService представляет сервис для обработки начислений.
+type AccrualService struct {
+	storage          accrualStorage         // Хранилище для работы с данными начислений.
+	jobQueueService  accrualJobQueueService // Сервис очереди задач для управления заданиями.
+	externalEndpoint string                 // Внешний endpoint для получения данных начислений.
+}
+
+// Интерфейс для хранилища начислений.
+type accrualStorage interface {
+	UpdateOrderStatus(ctx context.Context, orderID string, status database.OrderStatusDB) error
+	CreateAccrual(ctx context.Context, orderID string, amount float64) error
+	FindAllUnprocessedOrders(ctx context.Context) (*[]database.OrderDB, error)
+}
+
+// Интерфейс для сервиса очереди задач начислений.
+type accrualJobQueueService interface {
+	Enqueue(job Job)                          // Добавление задания в очередь.
+	ScheduleJob(job Job, delay time.Duration) // Планирование задания с задержкой.
+	PauseAndResume(delay time.Duration)       // Приостановка и возобновление обработки заданий.
+}
+
+// Конструктор для создания нового экземпляра AccrualService.
+func NewAccrualService(storage accrualStorage, jobQueueService accrualJobQueueService, externalEndpoint string) *AccrualService {
+	return &AccrualService{
+		storage:          storage,
+		jobQueueService:  jobQueueService,
+		externalEndpoint: externalEndpoint,
+	}
+}
+
+// CalculateAccrual инициирует процесс расчета начислений для указанного заказа.
+func (as *AccrualService) CalculateAccrual(orderID string) {
+	// Добавляем задание в очередь для обработки начислений.
+	as.jobQueueService.Enqueue(func(ctx context.Context) {
+		data, retryAfter, err := as.fetchAccrualData(orderID)
+
+		if err != nil {
+			if errors.Is(err, errNoOrder) {
+				logger.Log.Info("Заказ не зарегистрирован", zap.String("orderID", orderID))
+				// TODO: Возможно, нужно добавить повторное добавление задания в очередь.
+				return
+			}
+
+			logger.Log.Error("Не удалось получить данные начислений", zap.Error(err))
+			// TODO: Добавить повторное добавление задания с таймаутом.
+			return
+		}
+
+		if retryAfter > 0 {
+			logger.Log.Info("Получен Retry-After", zap.Int("retryAfter", retryAfter), zap.String("orderID", orderID))
+			as.jobQueueService.PauseAndResume(time.Second * time.Duration(retryAfter))
+			as.jobQueueService.Enqueue(func(ctx context.Context) {
+				as.CalculateAccrual(orderID)
+			})
+			logger.Log.Info("Добавлено новое задание после паузы", zap.Int("retryAfter", retryAfter), zap.String("orderID", orderID))
+			return
+		}
+
+		logger.Log.Info("Получены данные начислений",
+			zap.String("orderID", orderID),
+			zap.String("status", string(data.Status)),
+		)
```

Comment:
Так как мы в каждой записи используем `orderID` и `status` может быть удобно сразу создать logger  c этими данными чтобы в дальнейшем использовать его:
```suggestion
                logger := logger.With(zap.String("orderID", orderID), zap.String("status", string(data.Status))
	        logger.Info("Получены данные начислений")
```